---
title: "Phthalate_Covar_RNAseq_DataCleaning_Female_GAPPS"
author: "Samantha Lapehn, Mariana Parenti, and Alison Paquette"
date: "04/16/25"
output: html_document
editor_options: 
  chunk_output_type: console
---

## Females- Phthalate, Covariate, and RNAseq Data Cleaning- GAPPS
This code clean covariate data, RNAseq data, and phthalate data for a female subset of the sample resulting in a final N=109 for the female-stratified phthalate RNAseq analyses.

### Load packages
```{r load packages}
library(tidyverse)
library(ggVennDiagram)
library(here)
library(assertthat)
library(edgeR)
library(RColorBrewer)
library(org.Hs.eg.db)
library(biomaRt)
library(kableExtra) 
library(RUVSeq) 
library(ggrepel)
library(corrplot)
```

### Load Data
```{r load data}
#Load Exposure Data
PrenatalExposures<-read.csv("/Volumes/paquette_a/GAPPSGrant_PrelimData/Resubmission_July2024/NEW_RawData/PWG_CovariatePull_4-9-2024/req_05_biomarkers.csv")

#Load Covariates
full_covar<-read.csv("/Volumes/paquette_a/GAPPSGrant_PrelimData/Resubmission_July2024/NEW_RawData/PWG_CovariatePull_4_15_2024/req_05_data.csv")
                        
rownames(full_covar)<-(full_covar$pathways_id)

## load RNA seq
load(file="/Volumes/paquette_a/GAPPSGrant_PrelimData/Resubmission_July2024/NEW_RawData/gapps_20240325.Rdata")

### Load BWPW

BWPW<-read.csv("/Volumes/paquette_a/GAPPSGrant_PrelimData/Resubmission_July2024/intermediateData/all_birthweightZscores_fetalPlacentalRatios.csv")

BWPW$ratio[BWPW$fetal_placental_ratio>15]<-NA

### Load Diff Covar version to include gest age in data cleaning for Table
GestAge <- read.csv("/Volumes/paquette_a/GAPPSGrant_PrelimData/OriginalData/gapps_harmonization.csv")
GestAge <- GestAge %>%
  dplyr::select(pathways_id, h_birth_gestage)
```

### Covariate Cleaning

First, we need to attach the RNA-sequencing analysis batch to the covariate data. 

```{r add batch var}
## attach batches to covar
## match batches data to pathways_id 
## code from Alison's CRH analysis in CANDLE 
## CRH_PreliminaryDataPreprocessing_3102023.Rmd
batches$pathways_id <- colnames(counts$counts)

covar <- dplyr::full_join(full_covar, 
                          ## join batches
                          batches %>% dplyr::select(c(pathways_id, batch = Analysis)),
                          by = join_by(pathways_id))
```

Second, we need to remove the `8888` codings for NAs and convert them to `NA` that R can interpret.

```{r update NA coding}
covar <- covar %>%
   mutate(across(where(is.numeric), ~ na_if(., 8888)))
```
Note, we are not adjusting for hypertension or GDM here.

Now, filter based on exclusion criteria.

```{r filterMissingRNAseq}
dim(covar)

covar_rnaseq <- covar %>% 
  dplyr::distinct() %>%
  ## remove the duplicates in the fourth batch for now (arrange by batch, 
  ## then take only the first sample)
  dplyr::arrange(batch) %>%
  dplyr::group_by(pathways_id) %>%
  dplyr::slice(1) %>%
  dplyr::ungroup() %>%
  dplyr::filter(pathways_id %in% colnames(lengthScaledTPM$counts)) %>%
  ## also remove multifetal gestations and abruptions
  dplyr::filter(h_placental_abrupt == 0 | is.na(h_placental_abrupt) == TRUE) %>%
  dplyr::filter(h_multi_gest == 0 | is.na(h_multi_gest) == TRUE) %>%
  ## some issues with unreasonable maternal BMIs (= 0)
  dplyr::filter(h_m_prepreg_bmi > 0 | is.na(h_m_prepreg_bmi) == TRUE) %>%
  dplyr::filter(h_c_sex=="F")


dim(covar_rnaseq)
```

Now, let's factorize as needed. 

```{r var as factors}
covar_rnaseq <- within(covar_rnaseq, {
  protocol <- factor(protocol, labels = c("Old", "New"))
  mra_protocol <- factor(mra_protocol, labels = c("Old", "New"))
  site <- factor(site, labels = c("UW", "Swedish", "Yakima"))
  h_m_race <- factor(h_m_race, labels = c("White", "Black_AA", "Asian",
                                         "Multiple", "Other"))
  race_collapsed <- dplyr::if_else(h_m_race %in% c("Multiple"), 
                                   "Other", 
                                   h_m_race) %>% 
    factor() %>%
    relevel(ref = "White")
  h_m_ethn <- factor(h_m_ethn, labels = c("Not Hispanic", "Hispanic or Latino"))
  h_m_enroll_educ <- factor(h_m_enroll_educ, labels =  c("<HS", "HS graduate", "Graduated college/technical school", "Some graduate work or degree")) %>% ## college grad is largest group
    relevel(ref = "Graduated college/technical school")
  edu_collapsed <- dplyr::if_else(h_m_enroll_educ %in% c("Graduated college/technical school", "Some graduate work or degree"),
                                  0,  ## college graduate or higher as reference
                                  1) %>%
    factor()
  h_m_enroll_smoke <- factor(h_m_enroll_smoke, labels = c("No", "Yes"))
  cg_labor_type <- factor(cg_labor_type, labels = c("Spontaneous", "Augmented", "Induced", "No labor"))
  labor_collapsed <- dplyr::if_else(cg_labor_type == "No labor", 
                                    "No labor",
                                    "Labored") %>%
    factor(levels = c("Labored", "No labor"))
  h_del_method <- factor(h_del_method, labels = c("Vaginal", "C-section"))
  batch <- factor(batch)
  })
```

### Add Smoking and Gestage
```{r smoking and gestage}
covar_rnaseq <- full_join(covar_rnaseq, GestAge, by=c("pathways_id"))
dim(covar_rnaseq)
sum(duplicated(covar_rnaseq$pathways_id))
#Smoking- Making a composite variable where any Yes to self-report or any cotinine above threshold=Positive
Cotinine <- PrenatalExposures %>%
  dplyr::select(pathways_id, cotinine_yn, phthalate_contact_type)
Cotinine_T1 <- Cotinine %>%
  dplyr::filter(phthalate_contact_type=="T1")
Cotinine_T1 <- Cotinine_T1[-63, ] #There is a duplicate here for one participant with two T2 measurements, they are a 0 and NA so won't affect the results to remove one g1200502
summary(factor(Cotinine_T1$cotinine_yn)) # No T1 Smokers
Cotinine_T2 <- Cotinine %>%
  dplyr::filter(phthalate_contact_type=="T2")
Cotinine_T2 <- Cotinine_T2[-216, ] #There is a duplicate here for one participant with two T2 measurements, they are a 0 and NA so won't affect the results to remove one g1200502
summary(factor(Cotinine_T2$cotinine_yn)) # 5 T2 Smokers
Cotinine_T3 <- Cotinine %>%
  dplyr::filter(phthalate_contact_type=="T3")
summary(factor(Cotinine_T3$cotinine_yn)) # 4 T3 Smokers
#Make Cotinine composite by participant
Cotinine_T2_T3 <- full_join(Cotinine_T2, Cotinine_T3, by=c("pathways_id"), suffix=c("_T2", "_T3"))
Cotinine_AllTris<- full_join(Cotinine_T1, Cotinine_T2_T3, by=c("pathways_id"))
#Change 0s to NA
Cotinine_AllTris[is.na(Cotinine_AllTris)] <- 0
Cotinine_AllTris$CotinineComp <- Cotinine_AllTris$cotinine_yn + Cotinine_AllTris$cotinine_yn_T2 + Cotinine_AllTris$cotinine_yn_T3
summary(factor(Cotinine_AllTris$CotinineComp))
CotinineComp <- Cotinine_AllTris %>%
  dplyr::select(pathways_id, CotinineComp)
covar_rnaseq <- full_join(covar_rnaseq, CotinineComp, by=c("pathways_id"))

covar_rnaseq <- covar_rnaseq %>%
  dplyr::mutate(smoking_composite = dplyr::case_when(
  h_m_enroll_smoke == "Yes" ~ 1, ## if self-report, yes
  (h_m_enroll_smoke == "No" | is.na(h_m_enroll_smoke) == TRUE) & CotinineComp == 1 ~ 1,
  ## if self-report no or no report but cotinine == yes, yes
  h_m_enroll_smoke == "No" & (is.na(CotinineComp) == TRUE | CotinineComp == 0) ~ 0,
  ## if self-report no and cotinine == no or cotinine is NA, no
  is.na(h_m_enroll_smoke) == TRUE & (is.na(CotinineComp) == TRUE | CotinineComp == 0) ~ NA
  ## else, everything must be NA, so NA
))

covar_rnaseq$smoking_composite <- factor(covar_rnaseq$smoking_composite)
#Check Smoking Results
Smoking <- covar_rnaseq %>%
  dplyr::select(h_m_enroll_smoke, CotinineComp, smoking_composite)
summary(factor(covar_rnaseq$h_m_enroll_smoke))
summary(factor(covar_rnaseq$CotinineComp))
summary(factor(covar_rnaseq$smoking_composite))

dim(covar_rnaseq)
sum(duplicated(Cotinine_AllTris$pathways_id))
```


### Filter participants missing relevant covariates for each analysis

```{r na omit}
dim(covar_rnaseq)
covar_complete <- covar_rnaseq %>%
                                 dplyr::select(pathways_id, protocol, site, 
                                               h_c_sex, h_m_delivery_age, 
                                               race_collapsed, h_m_ethn, 
                                               edu_collapsed, h_m_prepreg_bmi, 
                                               parity_20, labor_collapsed, 
                                               h_del_method, batch, smoking_composite, h_birth_gestage) %>%
                                 na.omit()
                              

dim(covar_complete)
```

### Filter for only participants with complete covars
```{r filter phthalates}
Phthalate_Data1<-PrenatalExposures[,c(1,4,6,7:79)]
table(is.na(Phthalate_Data1$phthalate_contact_type))
Phthalate_Data<-Phthalate_Data1[!is.na(Phthalate_Data1$phthalate_contact_type),]

Phthalate_Data_Covars <- Phthalate_Data %>%
  dplyr::filter(pathways_id %in% covar_complete$pathways_id)

```

### Evaluate exposure data
```{r evaluate exp data}
#Compare Histogram of Visits- Updating trimesters here for 0-13w T1, 14-27w=T2, 28w+=T3
Phthalate_Data_Covars$Trimester<-cut(Phthalate_Data_Covars$gest_age_sample_days,
               breaks=c(0,7*14,7*28,max(Phthalate_Data_Covars$gest_age_sample_days)),
               right=T,labels=c("Tri1","Tri2","Tri3"))

T1Visit<-subset(Phthalate_Data_Covars,phthalate_contact_type=="T1")
table(T1Visit$Trimester)

T2Visit<-subset(Phthalate_Data_Covars,phthalate_contact_type=="T2")
table(T2Visit$Trimester)

T3Visit<-subset(Phthalate_Data_Covars,phthalate_contact_type=="T3")
table(T3Visit$Trimester)

hist(T1Visit$gest_age_sample_days,breaks=25,col="darkslategray1",xlim=c(0,300),ylim=c(0,10),main="Gestational Age at Sample collection",xlab="Gestational Age (Days)")
hist(T2Visit$gest_age_sample_days,breaks=50,col="cyan3",add=T)
hist(T3Visit$gest_age_sample_days,breaks=50,col="cyan4",add=T)
abline(v=c(98,196),lwd=4,col="red") #Marking trimesters 

#Venn Diagram to see how many participants have collection at each visit
T1 <- T1Visit$pathways_id
T2 <-T2Visit$pathways_id
T3 <- T3Visit$pathways_id
T_list <- list(T1, T2, T3)
ggVennDiagram(T_list, category.names=c("T1", "T2", "T3"))

length(unique(Phthalate_Data_Covars$pathways_id))
table(Phthalate_Data$phthalate_contact_type)
table(Phthalate_Data$Trimester)
```


### Process Phthalates
Use geometric mean to combine phthalate measures across multiple visits
```{r process phthalate data}
#Pull Only Phthalate Data
PhthalateOnlyData <-Phthalate_Data_Covars[, c(1, 5:26) ]

gm_mean <- function(a) {
  prod(a, na.rm = TRUE)^(1/length(a))
}

# Extract the grouping variable (pathways_id)
Gapps_IDs <- PhthalateOnlyData$pathways_id

# Aggregate using geomtric mean
PHT_Av <- aggregate(PhthalateOnlyData[, -which(names(PhthalateOnlyData) == "pathways_id")], 
                     by = list(pathways_id = Gapps_IDs), 
                     FUN = gm_mean)

rownames(PHT_Av)<-PHT_Av$pathways_id

#Pull Only LOD Data
LODs <-Phthalate_Data_Covars[, c(1, 27:48) ]
#Aggregate LODs as arithmetic averages
LODs_Agg <-aggregate(LODs[, -which(names(LODs) == "pathways_id")], 
                     by = list(pathways_id = Gapps_IDs), 
                     FUN = mean)
rownames(LODs_Agg) <- LODs_Agg$pathways_id
LODs_Agg <- LODs_Agg %>%
  dplyr::select(-pathways_id)
LODs_Factors <- sapply(LODs_Agg, as.factor)
Table<-data.frame(matrix(NA,nrow=22,ncol=5))
  colnames(Table)<-c("Total_0","Total_0.33", "Total_0.5", "Total_0.67", "Total_1")
  for(i in 1:22){
    Table[i,1]<-sum(LODs_Factors[,i]==0)
    Table[i,2]<-sum(LODs_Factors[,i]=="0.333333333333333"
)
     Table[i,3]<-sum(LODs_Factors[,i]==0.5
)
      Table[i,4]<-sum(LODs_Factors[,i]=="0.666666666666667"
)
    Table[i,5]<-sum(LODs_Factors[,i]==1)
  }
  rownames(Table)<-colnames(LODs_Factors)
  Table$SumParticipants <- (Table$Total_0 + Table$Total_0.5 + Table$Total_0.33 + Table$Total_0.67 + Table$Total_1)
  print(Table)
  
  Cutoff<-dim(LODs_Agg)[1]*0.7 # only retaining compounds with at least 70% of participants above LOD
  
 # Remove Samples Below LOD
  BelowLOD<-rownames(Table)[which(Table[,1]<Cutoff)]  #These Are the sample swe are removing
  print(BelowLOD)
  
  AboveLOD<-rownames(Table)[which(Table[,1]>Cutoff)]  #Samples We will retain
  AboveLOD<-str_remove(AboveLOD,"_lod")
  
   PHT<-PHT_Av[,AboveLOD]
  rownames(PHT)<-rownames(PHT_Av)
  PHT<-PHT[,-6] #removing phthalic acid (pa)
```

### Process Specific Gravity
Use geometric mean to process specific gravity measures across multiple visits
```{r process SG data}
#Pull Only SG  Data
SGData <-Phthalate_Data_Covars %>%
  dplyr::select(pathways_id, sg)

# aggregate using geometric mean
SG_Av <- aggregate(SGData[, -which(names(SGData) == "pathways_id")], 
                     by = list(pathways_id = Gapps_IDs), 
                     FUN = gm_mean)

rownames(SG_Av)<-SG_Av$pathways_id
colnames(SG_Av) <- c("pathways_id", "SG")
```


### Summarize Phthalte Data and Log Transform Phthalate data
```{r log transform}
#Plot raw values
boxplot(PHT, Title="Raw Phthalates (Average)")

#Summarize Raw Data
summary_rawdata <- summary(PHT)
print(summary_rawdata)

 #Log transform and visualize
    Log_PHT<-log(PHT) #the default is a natural log transformation
   boxplot(Log_PHT,Title="Natural Log Phthalates (Average)") 

```

### Make Phthalate Box Plots
```{r box plot phthalates}
#Step 1: Reorder dataframe by Median
PHT_sorted <-Log_PHT[order(-sapply(Log_PHT, median))]
colnames(PHT_sorted) <- toupper(colnames(PHT_sorted))
#Step 2: Make Boxplot
boxplot(PHT_sorted, cex.axis=1, col="dodgerblue4", ylim=c(-5,10), ylab="Natural Log Phthalate Concentration (ng/mL)")
stripchart(PHT_sorted,              # Data
           method = "jitter", # Random noise
           jitter=0.25,
           pch = 20, 
           cex=0.03,# Pch symbol
           col = "grey34",           # Color of the symbol
           add = TRUE, 
           vertical=TRUE)

```

### Make Phthalate Correlation Plots
This is using the pearson correlation which is default for corrplot on the log transformed phthalate data. 
Note: To move asterisks above numbers run the trace line below then edit line 443 to include +0.35 after sig.locs
trace(corrplot, edit=T)
```{r phthalate corr plots}
corrtest <- cor.mtest(PHT_sorted, conf.level = .95)
corrplot(cor(PHT_sorted),type="lower", method="color",order = "hclust", p.mat = corrtest$p,  addCoef.col = "black",insig = "label_sig", pch.cex=2, pch.col="red", number.cex=0.75, tl.col="black")

```

### Filter Covars to only those with phthalate data
Filtering covariates to only those with phthalate data then adding the specific gravity variable
```{r filter covars final}
covar_final <- covar_complete %>%
  dplyr::filter(pathways_id %in% rownames(PHT))

#check that order is the same
sum(covar_final$pathways_id == PHT_Av$pathways_id)

#add specific gravity
covar_final <- inner_join(covar_final, SG_Av, by=c("pathways_id"))
```

### Add BWPW to covars
Not planning to use this set since the number is so much lower (N=110 vs N=82), but just seeing what the overlap is with subset that has BWPW. 
```{r bwpw}
#filter to only include BWPW ratio and placental weight (birthweight already in covars)
BWPW_only <- BWPW %>%
  dplyr::select(pathways_id, m_pl1_weight, fetal_placental_ratio)

#Add to covars
covar_final <- inner_join(covar_final, BWPW_only, by=c("pathways_id"))

#full covars -no na sample size check
covar_check <- na.omit(covar_final) #N=81
```

### Check Trimesters for each compound
```{r check tris}
LODs_AllData<-Phthalate_Data[,grep("_lod", colnames(Phthalate_Data))]
LODs_AllData <- sapply(LODs_AllData, as.factor)
LODs_AllData <- data.frame(LODs_AllData)
LODs_AllData$phthalate_contact_type <- Phthalate_Data$phthalate_contact_type
LODs_AllData$pathways_id<- Phthalate_Data$pathways_id
Participants223 <- covar_final$pathways_id
T1_LODs <- LODs_AllData %>%
  dplyr::filter(phthalate_contact_type=="T1") 
T1_LODs <- T1_LODs %>%
  dplyr::filter(T1_LODs$pathways_id %in% Participants223) %>%
  dplyr::select(-pathways_id, -phthalate_contact_type)
T1_LODs <- mutate_if(T1_LODs, is.character, as.factor)
summary(T1_LODs)
T1_LOD_Summary <- summary(T1_LODs)

T2_LODs <- LODs_AllData %>%
  dplyr::filter(phthalate_contact_type=="T2") 
T2_LODs <- T2_LODs %>%
  dplyr::filter(T2_LODs$pathways_id %in% Participants223)%>%
  dplyr::select(-pathways_id, -phthalate_contact_type)
T2_LODs <- mutate_if(T2_LODs, is.character, as.factor)
summary(T2_LODs)
T2_LOD_Summary <- summary(T2_LODs)

T3_LODs <- LODs_AllData %>%
  dplyr::filter(phthalate_contact_type=="T3") 
T3_LODs <- T3_LODs %>%
  dplyr::filter(T3_LODs$pathways_id %in% Participants223)%>%
  dplyr::select(-pathways_id, -phthalate_contact_type)
T3_LODs <- mutate_if(T3_LODs, is.character, as.factor)
summary(T3_LODs)
T3_LOD_Summary <- summary(T3_LODs)

```

## Plot phthalates by Site
```{r}
PHT_forPlot <- stack(Log_PHT)
PHT_forPlot$pathways_id <- rep(rownames(Log_PHT), times = ncol(PHT))

Sites <- covar_final %>%
  dplyr::select(pathways_id, site)

PHT_Sites <- inner_join(PHT_forPlot, Sites, by=c("pathways_id"))

PHT_Sites <- PHT_Sites %>%
  mutate(city = recode(site, "UW" = "Seattle", "Swedish" = "Seattle", "Yakima" = "Yakima"))

colnames(PHT_Sites) <- c("LogConc", "Phthalate", "pathways_id", "Site", "City")
PHT_Sites$Phthalate <- toupper(PHT_Sites$Phthalate)
PHT_Sites$Phthalate <- reorder(PHT_Sites$Phthalate, PHT_Sites$LogConc, fun=median, decreasing=TRUE)

ggplot(PHT_Sites, aes(x=Phthalate, y=LogConc, fill=Site)) +
  geom_boxplot() + 
  theme_bw() + 
  ylab("Natural Log Concentration (ng/mL)") + 
  xlab("Phthalate Metabolite") + 
  ggtitle("Female-Phthalate Concentration by Site")

ggplot(PHT_Sites, aes(x=Phthalate, y=LogConc, fill=City)) +
  geom_boxplot() + 
  theme_bw() + 
  ylab("Natural Log Concentration (ng/mL)") + 
  xlab("Phthalate Metabolite") + 
  ggtitle("Female-Phthalate Concentration by City")
```



### RNA Sequencing Cleaning

Start by removing the duplicated sample and the NovoSeq results from the samples that were run on both instruments, using length scaled TPM values.

```{r pull data}
## remove the batch 4 duplicates
LS_Counts <- lengthScaledTPM$counts[, -ind[[2]]]

## check if duplicated names
colnames(LS_Counts)[duplicated(colnames(LS_Counts))]
dim(LS_Counts)

## remove duplicate
LS_Counts <- LS_Counts[, -which(duplicated(colnames(LS_Counts)))]

## confirm duplicate removed
colnames(LS_Counts)[duplicated(colnames(LS_Counts))]
dim(LS_Counts)
```

We will follow these steps in this order. 

1. Remove participants that don't have full covariate data
2. Removed Enzembl IDs with the suffix "_PAR_Y"  
3. Remove non-protein coding genes
4. Performed Filtering on genes with mean cpm<0  
5. Removed Duplicated Genes  
6. TMM Normalization 

```{r filtering}
## Original Dimensions
dim(LS_Counts)
dim(annot)

## Retain only complete cases (using covar_ratio) ####
LS_Counts <- LS_Counts[, covar_final$pathways_id]

## Dimensions after filtering for complete cases
dim(LS_Counts)

## Remove Ensembl IDs with the suffix "_PAR_Y" ####
LS_Counts <- LS_Counts[!grepl("_PAR_Y", rownames(LS_Counts)), ]
annot$ENSEMBL <- as.character(annot$ENSEMBL)
annot <- annot[!grepl("_PAR_Y",annot$ENSEMBL), ]

## Dimsenions after "_PAR_Y" removal
dim(LS_Counts)
dim(annot)

## get rownames of count dataset to match annot$ENSEMBL by removing the "." and 
## everything after it
rownames(LS_Counts) <- gsub("\\..*", "", rownames(LS_Counts))

## Retain only protein coding genes ####
## filter the annot file
annot <- subset(annot, BIOTYPE == "protein_coding") %>%
  droplevels()

## then filter based on what remains in annotation file
LS_Counts <- LS_Counts[as.character(annot$ENSEMBL), ]

## Dimensions retaining only protein coding genes
dim(LS_Counts)
dim(annot)

## Remove genes with low expression ####
## Keep genes with average log cpm>0
logcpm <- cpm(LS_Counts, log = T)

keep <- rowMeans(logcpm) > 0
summary(keep)

logcpm_filt <- logcpm[keep, ]
LS_Counts_filt<-LS_Counts[keep, ]
annot <- annot[keep, ]

## Plot before and after filtering out low expressing genes

plot(density(logcpm), main = "Before Removing low expressing genes")
abline(v = 0, col = "red", lty = 2)

plot(density(logcpm_filt), main = "After Removing low expressing genes")
abline(v = 0, col = "red", lty = 2)

## Dimensions after low expression filtering
dim(LS_Counts_filt)
dim(annot)
dim(logcpm_filt)

## Remove duplicated genes ####
summary(duplicated(logcpm_filt))

dupgenes <-as.character(annot[annot$ENSEMBL %in% rownames(logcpm_filt)[duplicated(logcpm_filt)], "SYMBOL"])
dupgenes

annot<-annot[-which(annot$SYMBOL%in%dupgenes),]
logcpm_filt<-logcpm_filt[-which(duplicated(logcpm_filt)),]
LS_Counts_filt<-LS_Counts_filt[-which(duplicated(LS_Counts_filt)),]

## Dimensions after removing duplicates
dim(LS_Counts_filt)
dim(logcpm_filt)
dim(annot)
```

Then conduct TMM normalization.

```{r TMM Norm}
#Make DGEList
Y <- DGEList(counts = LS_Counts_filt, genes = annot$SYMBOL)

Y_norm <- calcNormFactors(Y, method = "TMM") #This is the Default Method

#Plot before and after normalization
boxplot(cpm(Y ,log = T)[, 1:10], main = "Before Normalization")

boxplot(cpm(Y_norm, log = T)[, 1:10], main = "After Normalization")
```

### Covariate Summary
```{r covar summary}
summary(covar_final)
```

### Save Data
```{r save}
#write.csv(summary_rawdata, "/Volumes/paquette_a/GAPPSGrant_PrelimData/Sam/Data/Female/Female_Phthalate_RawDataSummary_GeoMean_N109_041625.csv")
#write.csv(PHT, "/Volumes/paquette_a/GAPPSGrant_PrelimData/Sam/Data/Female/Female_Phthalate_CleanedRawData_GeoMean_N109_041625.csv")
#write.csv(Log_PHT, "/Volumes/paquette_a/GAPPSGrant_PrelimData/Sam/Data/Female/Female_Phthalate_CleanedLogData_GeoMean_LN_N109_041625.csv")
#save(covar_final, file="/Volumes/paquette_a/GAPPSGrant_PrelimData/Sam/Data/Female/Female_FinalCovariates_N109_041625.Rdata")
#save(Y_norm, annot, file="/Volumes/paquette_a/GAPPSGrant_PrelimData/Sam/Data/Female/Female_FilteredNormalizedRNAseq_N109_041625.RData")
#write.csv(Table, "/Volumes/paquette_a/GAPPSGrant_PrelimData/Sam/Data/Female/Female_AboveLODNumbers_N109_041625.csv")
#write.csv(T1_LOD_Summary, "/Volumes/paquette_a/GAPPSGrant_PrelimData/Sam/Data/Female/Female_T1_LOD_Summary_041625.csv")
#write.csv(T2_LOD_Summary, "/Volumes/paquette_a/GAPPSGrant_PrelimData/Sam/Data/Female/Female_T2_LOD_Summary_041625.csv")
#write.csv(T3_LOD_Summary, "/Volumes/paquette_a/GAPPSGrant_PrelimData/Sam/Data/Female/Female_T3_LOD_Summary_041625.csv")
```

