---
title: "Male_GAPPS: Placental Gene Expression and Phthalate Exposures"
author: "Samantha Lapehn and Mariana Parenti"
date: "Current Version: `r format(Sys.time(), '%d %B %Y')`"
output: html_document
editor_options: 
  chunk_output_type: console
---


## Male-Stratified Differential Expression and Pathway Analysis for GAPPS Phthalates and Placental Transcriptome
This code performs the RNAseq analysis for associations between phthalates and the placental transcriptome in a male-stratified subset (N=113) of the sample.We used FDR<0.05 to define significant genes. 

### Load Libraries

```{r load packages}
library(tidyverse)
library(here)
library(assertthat)
library(edgeR)
library(RColorBrewer)
library(kableExtra) 
library(RUVSeq) 
library(ggrepel)
library(biomaRt)
library(org.Hs.eg.db)
```

#### Load Data

```{r load data}
## Filterd and Normalized RNAseq
load(file="/Volumes/paquette_a/GAPPSGrant_PrelimData/Sam/Data/Male/Male_FilteredNormalizedRNAseq_N113_041625.RData")

## Filtered and Natural Log Transformed Phthalate Data
PHT_noDEHP <-read.csv("/Volumes/paquette_a/GAPPSGrant_PrelimData/Sam/Data/Male/Male_Phthalate_CleanedLogData_GeoMean_LN_N113_041625.csv")

DEHP <- read.csv("/Volumes/paquette_a/GAPPSGrant_PrelimData/Sam/Data/DEHP_Clean_Male_N113_050825.csv")

PHT <- inner_join(PHT_noDEHP, DEHP, by=c("X"="pathways_id"))

rownames(PHT) <- PHT$X
PHT <- PHT %>%
  dplyr::select(-X, -X.y)

## Filtered covariates
load(file="/Volumes/paquette_a/GAPPSGrant_PrelimData/Sam/Data/Male/Male_FinalCovariates_N113_041625.Rdata")
covars <- covar_final
covars <- data.frame(covars)
rownames(covars) <- covars$pathways_id

sum(rownames(PHT) == rownames(covars))

#Combine Covar + Phthalates
covar_PHT <- merge(covars, PHT, by='row.names')

#Check variable types
summary(covar_PHT)
covar_PHT$batch <- factor(covar_PHT$batch, levels=c("1", "2", "3"))
summary(covar_PHT$batch)

```

### Differential Gene Expression Analysis

```{r DEGanalysis}
all_PHT_results <- lapply(colnames(PHT)[1:16], function(phthalate) {
    ## first, build mod using get function
  mod <- with(covar_PHT, 
              model.matrix(~ get(phthalate) + 
                               labor_collapsed +
                               h_del_method +
                               h_m_delivery_age + 
                              h_m_prepreg_bmi + 
                               h_m_ethn + 
                               race_collapsed + 
                               edu_collapsed +
                               parity_20 + 
                              SG + 
                             smoking_composite + 
                             batch
                             ))
  
  ## now, limma-voom
  v <- voom(Y_norm, mod, plot = TRUE)  
  designMatrix <- v$design
  fit <- eBayes(lmFit(v, designMatrix))

  ## get results (remember, we want coef 2) and turn into a tibble
  res_PHT <- topTable(fit, coef = 2, number = dim(fit)[1], adjust.method = "BH") %>%
    as_tibble(rownames = pkgconfig::get_config("tibble::rownames", "ensembl_id")) %>%
    dplyr::mutate(Phthalate = phthalate) %>%
    dplyr::relocate(Phthalate)
  
  }) %>%
  dplyr::bind_rows() 

lapply(colnames(PHT)[1:16], function(phthalate) {
  temp <- all_PHT_results %>%
    dplyr::filter(Phthalate == phthalate)
  
  tibble(Phthalate = phthalate, 
         `FDR<0.05` = sum(temp$adj.P.Val<0.05),
         `FDR<0.10` = sum(temp$adj.P.Val<0.1),
         `FDR<0.20` = sum(temp$adj.P.Val<0.2),
         `P<0.005` =  sum(temp$P.Value<0.005))
  }) %>%
  dplyr::bind_rows() %>%
  kable() %>%
  kableExtra::kable_styling()
```


### Plotting p-values
```{r fig.width=10, fig.height=10}
all_PHT_results$Phthalate <-toupper(all_PHT_results$Phthalate)
ggplot(all_PHT_results, aes(x = P.Value)) +
  geom_histogram(bins = 20, fill = "grey") +
  geom_hline(yintercept = 13156/20, linetype = "dashed", color = "tomato") +
  theme_classic() +
  facet_wrap(. ~ Phthalate, scales = "fixed")
```

### Make Phthalate DEG Table
3 DEGs for Males
```{r phthalate dEG table}
#Filter for signficant data
sig_PHT_results <- all_PHT_results %>%
  dplyr::filter(adj.P.Val<0.05)
unique(sig_PHT_results$Phthalate)
#Export to reformat as DEG table
#write.csv(sig_PHT_results,"/Volumes/paquette_a/GAPPSGrant_PrelimData/Sam/Data/Male/Phthalate_MaleDEGs_071025.csv")

```

### Make Phthalate Volcano Plots of Sig Metabolites only
Make a volcano plot for each phthalate metabolite with DEGs
```{r phthalate volcano plot}
#Step 1: Which phthalate metabolites have DEGs
sig_phthalates <- as.factor(sig_PHT_results$Phthalate)
summary(sig_phthalates)

#Step 2: Split Data for VPlots
mmp_allresults <- all_PHT_results %>%
  dplyr::filter(Phthalate=="MMP")


#Step 3: Volcano Plot Function
MakeVPlot<-function(Results,Title){
sig<-subset(Results,adj.P.Val<0.05)
sig<-sig[order(sig$adj.P.Val),]


sig_up<-subset(sig,logFC>0)

sig_down<-subset(sig,logFC<0)

ColorPalette<-brewer.pal(11,"RdBu")
ColorPalette<-ColorPalette[c(2:4,8:10)]

plot(Results$logFC,-log(Results$P.Value, base=10),ylim=c(0,6), xlim=c(-0.75, 0.75), pch=20,cex=0.35,main=Title,ylab="Negative Log p-value",xlab="Log Fold Change",col="grey60")
points(sig_up$logFC,-log(sig_up$P.Value, base=10),pch=20,cex=0.75,col=ColorPalette[3])

points(sig_down$logFC,-log(sig_down$P.Value, base=10),pch=20,cex=0.75,col=ColorPalette[4])

}
#Step 4: Make Volcano Plots
par(mfrow=c(1,1))
MakeVPlot(mmp_allresults, "MMP")

```

### Save all REsults
Saving unfiltered differential expression results
```{r save}
#write.csv(sig_PHT_results, "/Volumes/paquette_a/GAPPSGrant_PrelimData/Sam/Data/Male/MaleModel_DEGs_071025.csv")
#write.csv(all_PHT_results, "/Volumes/paquette_a/GAPPSGrant_PrelimData/Sam/Data/Male/MaleModel_AllResults_071025.csv")
```

