---
title: "High Dimensional Mediation Analysis with Placental Gene Expression"
author: "Mariana Parenti"
date: "Current Version: `r format(Sys.time(), '%d %B %Y')`"
output: html_document
---

## Aims

(a) Clean data for mediation analyses
(b) Evaluate the potential role for the placental transcriptome to act as a high-dimensional mediator between phthalate exposures and *i*) birthweight adjusted for placental weight (BW~adj~) or *ii*) birthweight:placental weight (BW:PW) ratio

## Covariate Data Cleaning

##### Load Libraries

```{r}
library(tidyverse)
library(here)
library(assertthat)
library(kableExtra)
library(arsenal)
library(HIMA)
library(edgeR)
library(splines)
```

#### Load Data

```{r}
## load RNA seq
load(here::here("Resubmission_July2024/NEW_RawData/gapps_20240325.Rdata"))

## load full covariate data
full_covar <- read_csv(file = here::here("Resubmission_July2024/NEW_RawData/PWG_CovariatePull_4_25_2024/req_05_data.csv"))

## load birthweight Z score data, which also has calculated placental_fetal_weight ratio
zscores <- read_csv(file = here::here("Resubmission_July2024", "intermediateData", "all_birthweightZscores_fetalPlacentalRatios.csv"))

## load this for cotinine values
load("/Volumes/paquette_a/GAPPSGrant_PrelimData/Resubmission_July2024/intermediateData/CleanedExposureData_AGP_April192024.RData")
rm(PAH, PHT)

## load exposure
## values are already log-transformed
PHTH <- read_csv(here::here("Sam/Data/Phthalate_CleanedLogData_GeoMean_LN_N222_041525.csv"))

DEHP <- read_csv(here::here("Mariana/clean_data/DEHP_Clean_N222_050825.csv"))

colnames(PHTH)[1] <- "pathways_id"

load("/Volumes/paquette_a/GAPPSGrant_PrelimData/Sam/Data/FinalCovariates_N222_041525.Rdata")
sg <- covar_final
rm(covar_final)

PHTH <- PHTH |>
  dplyr::full_join(sg |> dplyr::select(pathways_id, SG)) |>
  dplyr::full_join(DEHP |> dplyr::select(pathways_id, dehp = dehp2))
```

# Covariate Cleaning

First, we need to attach the RNA-sequencing analysis batch and birthweight z-scores to the covariate data. 

```{r}
## attach batches to covar
## match batches data to pathways_id 
## code from Alison's CRH analysis in CANDLE 
## CRH_PreliminaryDataPreprocessing_3102023.Rmd
batches$pathways_id <- colnames(counts$counts)

covar <- dplyr::full_join(full_covar, 
                          ## join batches
                          batches %>% dplyr::select(c(pathways_id, batch = Analysis)),
                          by = join_by(pathways_id)) %>%
  ## join fetal placental ratio from z-scores
  dplyr::full_join(zscores %>% dplyr::select(pathways_id, fetal_placental_ratio, 
                                             placenta_weight = m_pl1_weight, 
                                             birthweight = h_birthweight_g), 
                   by = join_by(pathways_id)) |>
  dplyr::full_join(Cotinine |> dplyr::select(pathways_id, cotinine, cotinine_yn),
                   by = join_by(pathways_id))
```

Second, we need to remove the `8888` codings for NAs and convert them to `NA` that R can interpret.

```{r}
covar <- covar %>%
   mutate(across(where(is.numeric), ~ na_if(., 8888)))
```

Third, we need to generate the cleaned hypertension and gestation diabetes variables. 

```{r}
covar_old <- covar %>%
  dplyr::filter(mra_protocol == 0) %>%
  dplyr::mutate(
    ## make chronic hypertension variable
    chronic_htn = dplyr::if_else(mra_old_m_category___12 == 1 | mra_old_m_medical_history___3 == 1, 1, 0),
    ## now make hypertensive disorders of pregnancy variable
    ## NAs from new protocol are retained
    htn_pregnancy = dplyr::if_else(h_preeclam_currpreg == 1 | h_gesthtn_currpreg == 1 | chronic_htn == 1,
                              1, 0),
    ## now make GDM variable
    gdm = dplyr::if_else(mra_old_m_diabete_type == 3 & is.na(mra_old_m_diabete_type) == FALSE, 1, 0)
  )

covar_new <- covar %>%
  dplyr::filter(mra_protocol == 1) %>%
  dplyr::mutate(
    ## make chronic hypertension variable
    chronic_htn = dplyr::if_else(mra_new_m_category___12 == 1, 1, 0),
    ## now make hypertensive disorders of pregnancy variable
    ## NAs from old protocol are retained
    htn_pregnancy = dplyr::if_else(h_preeclam_currpreg == 1 | h_gesthtn_currpreg == 1 | chronic_htn == 1,
                              1, 0),
    ## now make GDM variable
    gdm = dplyr::if_else(mra_new_m_category___53 == 1 & is.na(mra_new_m_category___53) == FALSE, 1, 0)
  )

covar_cleaned <- dplyr::bind_rows(covar_old, covar_new) 
```

Now, filter based on exclusion criteria.

```{r filterMissingRNAseq}
dim(covar_cleaned)

covar_rnaseq <- covar_cleaned %>% 
  ## n = 702
  dplyr::distinct() %>%
  ## n = 701
  ## remove the duplicates in the fourth batch for now (arrange by batch, 
  ## then take only the first sample)
  dplyr::arrange(batch) %>%
  dplyr::group_by(pathways_id) %>%
  dplyr::slice(1) %>%
  dplyr::ungroup() %>%
  ## n = 673
  dplyr::filter(pathways_id %in% colnames(lengthScaledTPM$counts)) %>%
  # n = 465
  ## also remove multifetal gestations and abruptions
  dplyr::filter(h_placental_abrupt == 0 | is.na(h_placental_abrupt) == TRUE) %>%
  ## n = 425; 451 when retaining NAs
  dplyr::filter(h_multi_gest == 0 | is.na(h_multi_gest) == TRUE) %>%
  ## n = 413; 438 when retaining NAs
  ## some issues with unreasonable maternal BMIs (= 0)
  dplyr::filter(h_m_prepreg_bmi > 0 | is.na(h_m_prepreg_bmi) == TRUE) |>
  ## n = 436
  dplyr::mutate(smoking_composite = dplyr::case_when(
    ## if self-report, yes
    h_m_enroll_smoke == 1 ~ 1, 
    ## if self-report no or no report but cotinine == yes, yes
    (h_m_enroll_smoke == 0 | is.na(h_m_enroll_smoke) == TRUE) & cotinine_yn == 1 ~ 1, 
    ## if self-report no and cotinine == no or cotinine is NA, no
    h_m_enroll_smoke == 0 & (is.na(cotinine_yn) == TRUE | cotinine_yn == 0) ~ 0, 
    ## else, everything must be NA, so NA
    is.na(h_m_enroll_smoke) == TRUE & (is.na(cotinine_yn) == TRUE | cotinine_yn == 0) ~ 8888
    ) |> na_if(8888)) 

dim(covar_rnaseq)
```

Now, let's factorize as needed. 

```{r}
covar_rnaseq <- within(covar_rnaseq, {
  protocol <- factor(protocol, labels = c("Old", "New"))
  mra_protocol <- factor(mra_protocol, labels = c("Old", "New"))
  site <- factor(site, labels = c("UW", "Swedish", "Yakima"))
  h_c_sex <- factor(h_c_sex) %>% ## relevel
    relevel(ref = "M") ## more males than females
  h_m_race <- factor(h_m_race, labels = c("White", "Black_AA", "Asian",
                                          "NI_AN", "Multiple", "Other"))
  race_collapsed <- dplyr::if_else(h_m_race %in% c("NI_AN", "Multiple"), 
                                   "Other", 
                                   h_m_race) %>% 
    factor() %>%
    relevel(ref = "White")
  h_m_ethn <- factor(h_m_ethn, labels = c("Not Hispanic", "Hispanic or Latino"))
  h_m_enroll_educ <- factor(h_m_enroll_educ, labels =  c("<HS", "HS graduate", "Graduated college/technical school", "Some graduate work or degree")) %>% ## college grad is largest group
    relevel(ref = "Graduated college/technical school")
  edu_collapsed <- dplyr::if_else(h_m_enroll_educ %in% c("Graduated college/technical school", "Some graduate work or degree"),
                                  0,  ## college graduate or higher as reference
                                  1) %>%
    factor()
  h_m_enroll_smoke <- factor(h_m_enroll_smoke, labels = c("No", "Yes"))
  smoking_composite <- factor(smoking_composite, labels = c("No", "Yes"))
  cg_labor_type <- factor(cg_labor_type, labels = c("Spontaneous", "Augmented", "Induced", "No labor"))
  labor_collapsed <- dplyr::if_else(cg_labor_type == "No labor", 
                                    "No labor",
                                    "Labored") %>%
    factor(levels = c("Labored", "No labor"))
  h_del_method <- factor(h_del_method, labels = c("Vaginal", "C-section"))
  gdm <- factor(gdm, labels = c("No", "Yes"))
  htn_pregnancy <- factor(htn_pregnancy, labels = c("No", "Yes"))
  batch <- factor(batch) ## batch 4 is largest, but no
  })
```

### Filter participants missing relevant covariates for each analysis

```{r}
covar_ratio <- covar_rnaseq %>%
  dplyr::select(pathways_id, protocol, site, batch, h_c_sex, h_m_delivery_age, 
                race_collapsed, h_m_ethn, edu_collapsed, h_m_prepreg_bmi, smoking_composite,
                parity_20, htn_pregnancy, gdm, labor_collapsed, h_del_method,
                fetal_placental_ratio, placenta_weight, birthweight,
                ## extras for table1
                h_birth_gestage, h_m_race, h_m_enroll_educ) %>%
  tidyr::drop_na(any_of(c("batch", "h_c_sex", "h_m_delivery_age", "smoking_composite",
                "race_collapsed", "h_m_ethn", "edu_collapsed", "h_m_prepreg_bmi", 
                "parity_20", "htn_pregnancy", "gdm", "labor_collapsed", 
                "h_del_method", "fetal_placental_ratio"))) |>
  dplyr::inner_join(PHTH, by = join_by(pathways_id)) |>
  droplevels()

dim(covar_ratio)
```

# Tables: Participant Characteristics

First, generate TABLE 1

```{r, results='asis'}
tab1 <- tableby( ~ h_m_delivery_age +
                   h_m_prepreg_bmi +
                   fetal_placental_ratio +
                   h_birth_gestage +
                   site +
                   batch +
                   h_c_sex +
                   labor_collapsed +
                   h_del_method +
                   smoking_composite +
                   h_m_enroll_educ +
                   h_m_race +
                   h_m_ethn,
                 data = covar_ratio, 
                 numeric.test = "kwt", cat.test = "chisq",
                 numeric.stats = c("median", "q1q3", "range", "sd", "Nmiss"),
                 cat.stats = c("countpct", "Nmiss"))

summary(tab1)
```

## Get spline knots

```{r}
covar_ratio$placenta_weight |> summary()

covar_ratio$placenta_weight |> scale() |> summary()

## calculated the 25, 50, and 75 percentile for scale(placenta_weight)
## in the N=163 participants included in this analysis
knots = c(-0.60980, 0.00497, 0.53831)
```

# RNA Sequencing Cleaning

Start by removing the duplicated sample and the NovoSeq results from the samples that were run on both instruments, using length scaled TPM values.

```{r}
## remove the batch 4 duplicates
LS_Counts <- lengthScaledTPM$counts[, -ind[[2]]]

## check if duplicated names
colnames(LS_Counts)[duplicated(colnames(LS_Counts))]
dim(LS_Counts)

## remove duplicate
LS_Counts <- LS_Counts[, -which(duplicated(colnames(LS_Counts)))]

## confirm duplicate removed
colnames(LS_Counts)[duplicated(colnames(LS_Counts))]
dim(LS_Counts)
```

We will follow these steps in this order. 

1. Remove participants that don't have full covariate data
2. Removed Enzembl IDs with the suffix "_PAR_Y"  
3. Remove non-protein coding genes
4. Performed Filtering on genes with mean cpm<0  
5. Removed Duplicated Genes  
6. TMM Normalization 

```{r filtering}
## Original Dimensions
dim(LS_Counts)
dim(annot)

## Retain only complete cases (using covar_ratio) ####
LS_Counts <- LS_Counts[, covar_ratio$pathways_id]

## Dimensions after filtering for complete cases
dim(LS_Counts)

## Remove Ensembl IDs with the suffix "_PAR_Y" ####
LS_Counts <- LS_Counts[!grepl("_PAR_Y", rownames(LS_Counts)), ]
annot$ENSEMBL <- as.character(annot$ENSEMBL)
annot <- annot[!grepl("_PAR_Y",annot$ENSEMBL), ]

## Dimsenions after "_PAR_Y" removal
dim(LS_Counts)
dim(annot)

## get rownames of count dataset to match annot$ENSEMBL by removing the "." and 
## everything after it
rownames(LS_Counts) <- gsub("\\..*", "", rownames(LS_Counts))

## Retain only protein coding genes ####
## filter the annot file
annot <- subset(annot, BIOTYPE == "protein_coding") %>%
  droplevels()

## then filter based on what remains in annotation file
LS_Counts <- LS_Counts[as.character(annot$ENSEMBL), ]

## Dimensions retaining only protein coding genes
dim(LS_Counts)
dim(annot)

## Remove genes with low expression ####
## Keep genes with average log cpm>0
logcpm <- cpm(LS_Counts, log = T)

keep <- rowMeans(logcpm) > 0
summary(keep)

logcpm_filt <- logcpm[keep, ]
LS_Counts_filt<-LS_Counts[keep, ]
annot <- annot[keep, ]

## Plot before and after filtering out low expressing genes

plot(density(logcpm), main = "Before Removing low expressing genes")
abline(v = 0, col = "red", lty = 2)

plot(density(logcpm_filt), main = "After Removing low expressing genes")
abline(v = 0, col = "red", lty = 2)

## Dimensions after low expression filtering
dim(LS_Counts_filt)
dim(annot)
dim(logcpm_filt)

## Remove duplicated genes ####
summary(duplicated(logcpm_filt))

dupgenes <-as.character(annot[annot$ENSEMBL %in% rownames(logcpm_filt)[duplicated(logcpm_filt)], "SYMBOL"])
dupgenes

annot<-annot[-which(annot$SYMBOL%in%dupgenes),]
logcpm_filt<-logcpm_filt[-which(duplicated(logcpm_filt)),]
LS_Counts_filt<-LS_Counts_filt[-which(duplicated(LS_Counts_filt)),]

## Dimensions after removing duplicates
dim(LS_Counts_filt)
dim(logcpm_filt)
dim(annot)
```

Then conduct TMM normalization.

```{r TMM Norm}
#Make DGEList
Y <- DGEList(counts = LS_Counts_filt, genes = annot$SYMBOL)

Y_norm <- calcNormFactors(Y, method = "TMM") #This is the Default Method

#Plot before and after normalization
boxplot(cpm(Y ,log = T)[, 1:10], main = "Before Normalization")

boxplot(cpm(Y_norm, log = T)[, 1:10], main = "After Normalization")
```

Write to file

```{r}
## write data to file 
save(covar_ratio, Y_norm, annot,
     file = here::here("Mariana", "output", "mediation_data.RData"))
```

# Get total effects for final table

### Test total effects of phthalates on birthweight

```{r}
exposures <- c("mep", "mcpp", "mehhp", "mbzp", "mmp", "mehp", "mhpp", "mcinp", 
               "mciop", "mbp", "mibp", "mecpp", "mcmhp", "mchpp", "meohp", "dehp")

te_table <- lapply(exposures, function(phthalate) {
  
  temp <- covar_ratio |>
    droplevels() |>
    ## now center and vectorize because glm doesn't like the attributes
    dplyr::mutate(
      #ratio = scale(fetal_placental_ratio) |> as.vector(),
      #birthweight = scale(birthweight) |> as.vector(),
      pw = scale(placenta_weight) |> as.vector(),
      h_m_prepreg_bmi = scale(h_m_prepreg_bmi) |> as.vector(),
      h_m_delivery_age = scale(h_m_delivery_age) |> as.vector(),
      parity_20 = scale(parity_20) |> as.vector(),
      SG = scale(SG) |> as.vector()
  )
  
  x <- covar_ratio[, phthalate] |> unlist() 
  
  mod <- lm(birthweight ~ 
              splines::ns(pw, knots=knots) + 
              h_c_sex + 
              h_m_delivery_age + 
              race_collapsed +  
              h_m_prepreg_bmi + 
              labor_collapsed + 
              h_del_method + 
              parity_20 +
              h_m_ethn + 
              edu_collapsed + 
              smoking_composite + 
              SG + 
              x,
            temp)
  
  broom::tidy(mod) |>
    dplyr::mutate(exposure = phthalate)
}) |>
  dplyr::bind_rows() |>
  dplyr::filter(term == "x") |>
  dplyr::select(-term) |>
  dplyr::relocate(exposure) 

te_table |>
  dplyr::arrange(p.value) |>
  kable() |>
  kableExtra::kable_styling()
```

### Total effect for ratio

```{r}
te_ratio_table <- lapply(exposures, function(phthalate) {
  
  temp <- covar_ratio |>
    droplevels() |>
    ## now center and vectorize because glm doesn't like the attributes
    dplyr::mutate(
      #ratio = scale(fetal_placental_ratio) |> as.vector(),
      #birthweight = scale(birthweight) |> as.vector(),
      pw = scale(placenta_weight) |> as.vector(),
      h_m_prepreg_bmi = scale(h_m_prepreg_bmi) |> as.vector(),
      h_m_delivery_age = scale(h_m_delivery_age) |> as.vector(),
      parity_20 = scale(parity_20) |> as.vector(),
      SG = scale(SG) |> as.vector()
  )
  
  x <- covar_ratio[, phthalate] |> unlist() 
  
  mod <- lm(fetal_placental_ratio ~ h_c_sex + labor_collapsed + 
              h_del_method +smoking_composite + h_m_delivery_age + 
              h_m_prepreg_bmi + h_m_ethn + race_collapsed +  edu_collapsed + 
              parity_20 + SG + x,
            temp)
  
  broom::tidy(mod) |>
    dplyr::mutate(exposure = phthalate)
}) |>
  dplyr::bind_rows() |>
  dplyr::filter(term == "x") |>
  dplyr::select(-term) |>
  dplyr::relocate(exposure) 

te_ratio_table |>
  dplyr::arrange(p.value) |>
  kable() |>
  kableExtra::kable_styling()
```

# High Dimensional Mediation Analysis

```{r}
## generate the scaled, centered covariates
covariates <- covar_ratio |>
  droplevels() |>
  ## now center and vectorize because glm doesn't like the attributes
  dplyr::mutate(
    placenta_weight = scale(placenta_weight) |> as.vector(),
    h_m_prepreg_bmi = scale(h_m_prepreg_bmi) |> as.vector(),
    h_m_delivery_age = scale(h_m_delivery_age) |> as.vector(),
    parity_20 = scale(parity_20) |> as.vector(),
    SG = scale(SG) |> as.vector()
  )

## exposures, centered and scaled
exposure_scaled <- covar_ratio |> 
  dplyr::select(c(mep:meohp, dehp)) |>
  scale(center = FALSE, scale = FALSE)

## scaled centered mediators, M
M_scaled <- Y_norm |> 
  cpm(log = TRUE) |>
  t() |>
  scale()

## covariates (not include HTN/GDM because on pathway?)
cov_mat <- with(covariates, model.matrix(
  ~ h_c_sex + 
    labor_collapsed +
    h_del_method +
    batch +
    smoking_composite + 
    h_m_delivery_age + 
    h_m_prepreg_bmi + 
    h_m_ethn + 
    race_collapsed + 
    edu_collapsed +
    parity_20 + 
    gdm +
    htn_pregnancy +
    SG
                                           )) |>
  as.data.frame() |>
  dplyr::select(-1)

set.seed(2024)

hima_res <- lapply(exposures, function(phthalate) {
  
  eHIMA.fit <- hima_efficient(
    X = exposure_scaled[, phthalate] |> unlist(),
    M = M_scaled,
    Y = covariates$fetal_placental_ratio |> unlist(),
    COV = cov_mat,
    scale = FALSE,
    FDRcut = 0.05,
    verbose = TRUE
    )

  ## print results
  temp <- eHIMA.fit |>
    as_tibble() |>
    dplyr::mutate(exposure = phthalate)
  
  print(temp)
  
  }) 

hima_res |>
  dplyr::bind_rows() |>
  dplyr::inner_join(annot |> dplyr::select(Index = ENSEMBL, gene_name = SYMBOL)) |>
  dplyr::relocate(gene_name) |>
  dplyr::arrange(gene_name) |>
  dplyr::select(-Index) |>
  kable() |>
  kableExtra::kable_styling() |>
  kableExtra::collapse_rows(target = 1)
```

Combine with total effect

```{r}
hima_res |>
  dplyr::bind_rows() |>
  dplyr::inner_join(annot |> dplyr::select(Index = ENSEMBL, gene_name = SYMBOL)) |>
  dplyr::full_join(te_ratio_table |>
                     dplyr::select(exposure, TE = estimate, TE_p = p.value), 
                   by = join_by(exposure)) |>
  dplyr::mutate(proportion_mediated = 100*(IDE/TE) |> round(3),
                exposure = stringr::str_to_upper(exposure),
                Index = substring(Index, 3)) |>
  dplyr::arrange(exposure, Index) |>
  dplyr::filter(TE_p<0.1) |>
  dplyr::select(exposure, gene_name, alpha_hat, beta_hat, IDE, proportion_mediated, pmax) |>
  kable() |>
  kableExtra::kable_styling() |>
  kableExtra::collapse_rows(target = 1)
```

Also birthweight (adjusted for the nonlinear term for placental weight)

```{r}
## covariates (not include HTN/GDM because on pathway?)
cov_mat <- with(covariates, model.matrix(
  ~ splines::ns(placenta_weight, knots=knots) +
    h_c_sex + 
    labor_collapsed +
    h_del_method +
    batch +
    smoking_composite + 
    h_m_delivery_age + 
    h_m_prepreg_bmi + 
    h_m_ethn + 
    race_collapsed + 
    edu_collapsed +
    parity_20 + 
    gdm +
    htn_pregnancy +
    SG
                                           )) |>
  as.data.frame() |>
  dplyr::select(-1)

set.seed(2024)

hima_res_bw <- lapply(exposures, function(phthalate) {
  
  eHIMA.fit <- hima_efficient(
    X = exposure_scaled[, phthalate] |> unlist(),
    M = M_scaled,
    Y = covariates$birthweight |> unlist(),
    COV = cov_mat,
    scale = FALSE,
    FDRcut = 0.05,
    verbose = TRUE
    )

  ## print results
  temp <- eHIMA.fit |>
    as_tibble() |>
    dplyr::mutate(exposure = phthalate)
  
  print(temp)
  
  }) 

hima_res_bw |>
  dplyr::bind_rows() |>
  dplyr::inner_join(annot |> dplyr::select(Index = ENSEMBL, gene_name = SYMBOL)) |>
  dplyr::relocate(gene_name) |>
  dplyr::arrange(gene_name, exposure) |>
  dplyr::select(-Index) |>
  kable() |>
  kableExtra::kable_styling() |>
  kableExtra::collapse_rows(target = 1)

hima_res_bw |>
  dplyr::bind_rows() |>
  dplyr::inner_join(annot |> dplyr::select(Index = ENSEMBL, gene_name = SYMBOL)) |>
  dplyr::relocate(gene_name) |>
  dplyr::arrange(exposure, gene_name) |>
  dplyr::relocate(exposure) |>
  dplyr::select(-Index) |>
  kable() |>
  kableExtra::kable_styling() |>
  kableExtra::collapse_rows(target = 1)
```


Combine with total effect

```{r}
hima_res_bw |>
  dplyr::bind_rows() |>
  dplyr::inner_join(annot |> dplyr::select(Index = ENSEMBL, gene_name = SYMBOL)) |>
  dplyr::full_join(te_table |>
                     dplyr::select(exposure, TE = estimate, TE_p = p.value), 
                   by = join_by(exposure)) |>
  dplyr::mutate(proportion_mediated = 100*(IDE/TE) |> round(3),
                exposure = stringr::str_to_upper(exposure)) |>
  dplyr::arrange(exposure, gene_name) |>
  dplyr::filter(TE_p<0.1) |>
  dplyr::select(exposure, gene_name, alpha_hat, beta_hat, IDE, proportion_mediated, pmax) |>
  kable() |>
  kableExtra::kable_styling() |>
  kableExtra::collapse_rows(target = 1)
```